src/Controller.ts

```diff
@@ -20,47 +18,36 @@ export class Controller {
         return this._manager.getAppList();
     }
 
-    public async startBootApp(app: BootApp): Promise<void> {
-        this._setState(app, STATE_RUNNING);
-        const outputChannel: vscode.OutputChannel = this._getOutput(app);
-        const classpathString = app.classpath.entries.map(e => e.kind === "source" ? e.outputFolder : e.path).join(path.delimiter);
+    public async startBootApp(app: BootApp, debug?: boolean): Promise<void> {
+        const mainClasData = await this._getMainClass(app.path);
 
-        // Note: Command `vscode.java.resolveMainClass` is implemented in extension `vscode.java.resolveMainClass`
-        const mainClassList = await vscode.commands.executeCommand('java.execute.workspaceCommand', 'vscode.java.resolveMainClass', app.path);
-        if (mainClassList && mainClassList instanceof Array && mainClassList.length > 0) {
-            const mainClassData: MainClassData = mainClassList.length === 1 ? mainClassList[0] :
-                await vscode.window.showQuickPick(mainClassList.map(x => Object.assign({ title: x.mainClass }, x)));
+        if (mainClasData) {
+            let targetConfig = this._getLaunchConfig(mainClasData);
+            if (!targetConfig) {
+                targetConfig = await this._createNewLaunchConfig(mainClasData);
+            }
+            // TO REMOVE:
+            // This is a workaround to make "start" work. For Java Debugger, "Start without debugging" feature 
+            // has not been released, it fallbacks to "Start debugging" mode.
+            // See: https://github.com/Microsoft/vscode-java-debug/issues/351
+            debug ? await this._enableAllBPs() : await this._disableAllBPs();
 
-            outputChannel.clear();
-            outputChannel.show();
-            let stderr: string = '';
-            const javaProcess: ChildProcess = spawn("java", ["-classpath", classpathString, mainClassData.mainClass]);
-            javaProcess.stdout.on('data', (data: string | Buffer): void => {
-                outputChannel.append(data.toString());
-            });
-            javaProcess.stderr.on('data', (data: string | Buffer) => {
-                stderr = stderr.concat(data.toString());
-                outputChannel.append(data.toString());
-            });
-            javaProcess.on('error', (err: Error) => {
-                console.error("on error: ", err.toString());
-            });
-            javaProcess.on('exit', (code: number, signal: string) => {
-                this._setState(app, STATE_INACTIVE);
-            });
-            app.process = javaProcess;
+            const ok: boolean = await vscode.debug.startDebugging(
+                vscode.workspace.getWorkspaceFolder(vscode.Uri.parse(app.path)),
+                Object.assign({}, targetConfig, { noDebug: !debug })
+            );
+            if (ok) {
+                app.activeSession = vscode.debug.activeDebugSession;
```

Comment:
If there is a debug session running, when I try to start a new one but finally click "Abort",  `vscode.debug.startDebugging` still returns true, the previous running debug session is  still "vscode.debug.activeDebugSession", so it is wrongly set as current app's  activeSession.